### PR TITLE
Syncronized coins in game instance

### PR DESCRIPTION
**Pull request by Warhedgehog**
Implemented methods to keep the coins of a game synchronized with the Database.
Namely, upon Game instance creation it will keep the list of coins updated with the database. Furthermore one can add a listener for the list of coins

### CODE REVIEW

File: app/src/main/java/sdp/moneyrun/game/Game.java [hunk 1]
```diff
@@ -209,6 +220,16 @@ public static Game getGameFromTaskSnapshot(Task<DataSnapshot> task){
         }
     }
 
+    public void addCoinListener(ValueEventListener listener){
+        if (listener == null) throw new IllegalArgumentException();
+        FirebaseDatabase.getInstance().getReference().child("open_games").child("coins").addValueEventListener(listener);
```
Comment (RafaelRamirezS) [inline app/src/main/java/sdp/moneyrun/game/Game.java:225]: I think youre on the wrong node. Did you intend to go into [...] .child("open_games").child(*GAMEID*).child("coins") ?
Or are the coins situated in the root "open_games" branch, and not in a Game in particular ?

File: app/src/main/java/sdp/moneyrun/game/Game.java [hunk 2]
```diff
@@ -209,6 +220,16 @@ public static Game getGameFromTaskSnapshot(Task<DataSnapshot> task){
...
+    }
+
+    public void removeCoinListener(ValueEventListener listener){
+        if (listener == null) throw new IllegalArgumentException();
+        FirebaseDatabase.getInstance().getReference().child("open_games").child("coins").removeEventListener(listener);
```
Comment (RafaelRamirezS) [inline app/src/main/java/sdp/moneyrun/game/Game.java:230]: Same comment as for addCoinListener, seeing as you probably copy pasted ?

**Review by RafaelRamirezS — CHANGES_REQUESTED** [review]

File: app/src/main/java/sdp/moneyrun/game/Game.java (hunk 1 above)
Comment (RafaelRamirezS) [reply app/src/main/java/sdp/moneyrun/game/Game.java:225]: Actually Arnau changed it from "open_games" to just "games" so you should change that too

File: app/src/main/java/sdp/moneyrun/game/Game.java (hunk 2 above)
Comment (RafaelRamirezS) [reply app/src/main/java/sdp/moneyrun/game/Game.java:230]: same as above, change "open_games" to "games"

File: app/src/main/java/sdp/moneyrun/game/Game.java (hunk 1 above)
Comment (Warhedgehog) [reply app/src/main/java/sdp/moneyrun/game/Game.java:225]: Thanks for the review you were right

**Comment by Warhedgehog** [comment]
Should be ready for merge. The previous change request have been addressed

**Review by Alvaro2112 — APPROVED** [review]

**Review by RafaelRamirezS — APPROVED** [review]
Requested changes were fixed, otherwise, all good

**Pull request merged**